### PR TITLE
Fix desi_resubmit_queue_failures corner cases

### PR DESCRIPTION
**Pull request by sbailey**
This PR fixes two bugs/features with desi_resubmit_queue_failures:
* #2618 : placeholder QID=1 was not excluded from sbatch dependency list
* Don't resubmit a job if the dependencies are known to be in a bad state

Also: I temporarily turned off the coveralls upload since they are having a major outage resulting in false-positive test failures.

The logic here is a bit subtle with lots of corner cases, so I'd appreciate a careful review from @akremin .  I'll explicitly add you as a reviewer after iterating with copilot and addressing any test failures.

## Further details: QID=1

The previous logic in `desispec.workflow.processing.submit_batch_script` seemed to be checking for values of None and 0, but not 1.  It also seemed to have leftover logic covering the possibility that dep_qids was a scalar instead of an array of ints, which I don't think happens, but maybe does in a corner case (e.g. if all jobs have a single dependency instead of multiple?)  The refactor assumes array of int, and checks for both 0 and 1 but not None.

## Further details: job resubmission

At each recursion level, after attempting to re-submit dependencies, this PR now checks that they actually succeeded before submitting the current job.  It also propagates the number of failures upward so that the log message of the script can report failures.  With m1 production and  current main:
```
$> desi_resubmit_queue_failures --dry-run-level 3 -n 20240627
[various error messages about failed dependencies on other nights, while also attempting to submit jobs anyway]
INFO:desi_resubmit_queue_failures:113:<module>: Completed all necessary queue resubmissions from processing table: /global/cfs/cdirs/desi/spectro/redux/m1/processing_tables/processing_table_m1-20240627.csv
```

with this PR
```
$> desi_resubmit_queue_failures --dry-run-level 3 -n 20240627
[various error messages about failed dependencies on other nights, with no attempted resubmissions of this night]
ERROR:desi_resubmit_queue_failures:115:<module>: Resubmitted 0 jobs but failed to submit 23 jobs for 20240627.
```

### CODE REVIEW

File: py/desispec/workflow/processing.py [hunk 1]
```diff
@@ -1487,9 +1495,35 @@ def recursive_submit_failed(rown, proc_table, submits, id_to_row_map, max_resubs
...
+            proc_table['STATUS'][rown] = "DEP_NOT_SUBD"
+            ok = False
+
+    if ok:
+        proc_table[rown] = submit_batch_script(proc_table[rown], reservation=reservation,
+                                               strictly_successful=True, dry_run=dry_run_level)
+        submits += 1
+    else:
+        log.error(f"Not submitting {row['INTID']} since dependencies aren't in good states.")
+        proc_table['STATUS'][rown] = "DEP_NOT_SUBD"  # just in case
```
[the remark this answers is not in the document]
Comment (sbailey) [reply py/desispec/workflow/processing.py]: updated fragile logic

File: py/desispec/workflow/processing.py
[the remark this answers is not in the document]
Comment (sbailey) [reply py/desispec/workflow/processing.py]: Thanks. I think I agree with your suggested change to simplify the logic, but will punt on implementing + retesting this until I have more time to evaluate it carefully.
